websites/R/Roblox/presence.ts

```diff
@@ -1,252 +1,713 @@
-	} else if (
-		(document.location.pathname === "/games/" ||
-			document.location.pathname === "/games") &&
-		!gameName
-	) {
-		presenceData.details = "Browsing games...";
+			presenceData.details = `Game: ${gameName.innerText}`;
+			presenceData.state = `Tab: ${gameTab.innerText}`;
+			presenceData.largeImageKey = imagesEnabled ? gameImage : "lg";
+			presenceData.startTimestamp = browsingStamp;
+
+			if (buttons) {
+				presenceData.buttons = [
+					{
+						label: "Visit Game",
+						url: document.URL
+					}
+				];
+			}
+		} else if (pathname.includes("/catalog")) {
+			const searchURL = new URL(document.location.href),
+				searchResult = searchURL.searchParams.get("Keyword");
+
+			presenceData.details = "Current page:";
+			presenceData.state = "Catalog";
+			presenceData.startTimestamp = browsingStamp;
+
+			const itemName = <HTMLElement>(
+					document.querySelector(".item-name-container h2")
+				),
+				itemImage = <HTMLImageElement>(
+					document.querySelector("span.thumbnail-span img")
+				);
+
+			if (searchResult) {
+				presenceData.details = "Searching for an item: ";
+				presenceData.state = searchResult;
+			} else if (itemImage) {
+				presenceData.details = "Looking at Catalog Item:";
+				(presenceData.largeImageKey = imagesEnabled ? itemImage.src : "lg"),
+					(presenceData.state = itemName.innerText);
+
+				if (buttons) {
+					presenceData.buttons = [
+						{
+							label: "View Catalog Item",
+							url: document.URL
+						}
+					];
+				}
+			}
+		} else if (pathname.includes("/places/")) {
+			const selectedTab = <HTMLDivElement>(
+				document.querySelector("#MasterContainer #navbar div.selected a")
+			);
+
+			presenceData.details = "Configuring Place";
+			presenceData.state = `Tab: ${selectedTab.innerText || "Unknown"}`;
+			presenceData.startTimestamp = browsingStamp;
+		} else if (pathname.includes("/universes/configure")) {
+			const selectedTab = <HTMLDivElement>(
+				document.querySelector("#MasterContainer #navbar div.selected a")
+			);
+
+			presenceData.details = "Configuring Experience";
+			presenceData.state = `Tab: ${selectedTab.innerText || "Unknown"}`;
+			presenceData.startTimestamp = browsingStamp;
+		} else if (pathname.includes("/bundles/")) {
+			const itemName = <HTMLElement>(
+					document.querySelector(".item-name-container h2")
+				),
+				itemImage = <HTMLImageElement>(
+					document.querySelector("span.thumbnail-span img")
+				);
+
+			presenceData.details = "Looking at Bundle:";
+			(presenceData.largeImageKey = imagesEnabled ? itemImage.src : "lg"),
+				(presenceData.state = itemName.innerText);
+			presenceData.startTimestamp = browsingStamp;
+
+			if (buttons) {
+				presenceData.buttons = [
+					{
+						label: "View Bundle",
+						url: document.URL
+					}
+				];
+			}
+		} else if (pathname.includes("/search/users")) {
+			const searchURL = new URL(document.location.href),
+				searchResult = searchURL.searchParams.get("keyword");
```

Comment:
ANd you can remove L324 in this case
```suggestion
				presenceData.state = searchURL.searchParams.get("keyword");
```